src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs

```diff
@@ -107,5 +107,48 @@ public void ConvertToInvariantNamesNoNames()
 
             Assert.Equal("If(B, Num, 1234)", displayExpressions);
         }
+
+        [Theory]
+        [InlineData("If(B, Num, 1234)", "If(A, Num, 1234)", "B", "A")]
+        [InlineData("RecordNest.SomeString", "RecordNest.SomeString", "B", "A")]
+        [InlineData("RecordNest.SomeString", "RecordNest.Foo", "RecordNest.SomeString", "Foo")]
+        [InlineData("RecordNest.SomeString", "Foo.SomeString", "RecordNest", "Foo")]
+        [InlineData("First(Nested).Inner", "First(Nested).Inner", "B", "A")]
+        [InlineData("First(Nested).Inner", "First(Nested).Bar", "Nested.Inner", "Bar")]
+        [InlineData("First(Nested).Inner", "First(Bar).Inner", "Nested", "Bar")]
+        [InlineData("First(Nested).InnerDisplay", "First(Bar).InnerDisplay", "Nested", "Bar")]
+        [InlineData("With({SomeValue: 123}, RecordNest.nest2.datetest)", "With({SomeValue: 123}, RecordNest.nest2.Foo)", "RecordNest.nest2.datetest", "Foo")]
+        [InlineData("With({RecordNest: {nest2: {datetest: 123}}}, RecordNest.nest2.datetest)", "With({RecordNest: {nest2: {datetest: 123}}}, RecordNest.nest2.datetest)", "RecordNest.nest2.datetest", "Foo")]
+        [InlineData("With(RecordNest, SomeString + nest2.datetest)", "With(Foo, SomeString + nest2.datetest)", "RecordNest", "Foo")]
+        [InlineData("With(RecordNest, SomeString + nest2.datetest)", "With(RecordNest, Foo + nest2.datetest)", "RecordNest.SomeString", "Foo")]
+        [InlineData("With(RecordNest, SomeString + nest2.datetest)", "With(RecordNest, SomeString + Foo.datetest)", "RecordNest.nest2", "Foo")]
+        [InlineData("With(RecordNest, SomeString + nest2.datetest)", "With(RecordNest, SomeString + nest2.Foo)", "RecordNest.nest2.datetest", "Foo")]
+        [InlineData("With({value: RecordNest.SomeString}, value & B)", "With({value: RecordNest.'abcd efg'}, value & B)", "RecordNest.SomeString", "abcd efg")]
+        public void RenameParameter(string expressionBase, string expectedExpression, string path, string newName)
```

Comment:
>path [](http://example.com/codeflow?start=93&length=4)

we take a path, which is based on instance (not type) , there may be some interesting tests here...

RecordNest.SomeString --> S2

With({ x : RecordNest, y : RecordNest}, 
   x.SomeString & y.SomeString)